_pytest/warnings.py

```diff
@@ -73,7 +73,7 @@ def catch_warnings_for_item(item):
 
             if compat._PY2 and any(isinstance(m, compat.UNICODE_TYPES) for m in warn_msg.args):
                 new_args = [compat.safe_str(m) for m in warn_msg.args]
-                unicode_warning = warn_msg.args != new_args
+                unicode_warning = list(warn_msg.args) != new_args
```

Comment:
It was always issue the "unicode message non-ascii compatible" because `warns_msg.args` is a `tuple`, which always compares as different than a `list`.